packages/playwright-core/src/server/dispatchers/elementHandlerDispatcher.ts

```diff
@@ -48,6 +49,8 @@ export class ElementHandleDispatcher extends JSHandleDispatcher<FrameDispatcher>
     const elementHandle = handle.asElement();
     if (!elementHandle)
       return new JSHandleDispatcher(scope, handle);
+    if (scope instanceof PageDispatcher)
```

Comment:
I think line 45 has a bug - we can't allow creating ElementHandleDispatcher with the JSHandleDispatcherParentScope, it should be ElementHandledispatcherParentScope, which strictly is FrameDispatcher, but you could probably allow PageDispatcher for convenience. That way you can remove this exception and also remove wrong code from electronDispatcher.